clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceChangelogConsumerImpl.java

```diff
@@ -224,75 +228,90 @@ protected VeniceCompressor getVersionCompressor(TopicPartition topicPartition) {
   }
 
   @Override
   public CompletableFuture<Void> seekToEndOfPush() {
     return seekToEndOfPush(
-        kafkaConsumer.assignment()
+        pubSubConsumer.getAssignment()
             .stream()
-            .map(topicPartition -> topicPartition.partition())
+            .map(topicPartition -> topicPartition.getPartitionNumber())
             .collect(Collectors.toSet()));
   }
 
   @Override
   public CompletableFuture<Void> seekToTail(Set<Integer> partitions) {
-    // Get latest change capture topic
-    String topic = getCurrentServingVersionTopic() + ChangeCaptureView.CHANGE_CAPTURE_TOPIC_SUFFIX;
-    return internalSeek(partitions, topic, kafkaConsumer::seekToEnd);
+    // Get the latest change capture topic
+    PubSubTopic topic =
+        pubSubTopicRepository.getTopic(getCurrentServingVersionTopic() + ChangeCaptureView.CHANGE_CAPTURE_TOPIC_SUFFIX);
+    return internalSeek(partitions, topic, p -> {
+      Long partitionEndOffset = pubSubConsumer.endOffset(p);
+      pubSubConsumerSeek(p, partitionEndOffset);
+    });
   }
 
-  private String getCurrentServingVersionTopic() {
+  private PubSubTopic getCurrentServingVersionTopic() {
     storeRepository.refresh();
     Store store = storeRepository.getStore(storeName);
     int currentVersion = store.getCurrentVersion();
-    return Version.composeKafkaTopic(storeName, currentVersion);
+    return pubSubTopicRepository.getTopic(Version.composeKafkaTopic(storeName, currentVersion));
   }
 
   @Override
   public CompletableFuture<Void> seekToTail() {
     return seekToTail(
-        kafkaConsumer.assignment()
+        pubSubConsumer.getAssignment()
             .stream()
-            .map(topicPartition -> topicPartition.partition())
+            .map(topicPartition -> topicPartition.getPartitionNumber())
             .collect(Collectors.toSet()));
   }
 
   @Override
   public CompletableFuture<Void> seekToCheckpoint(Set<VeniceChangeCoordinate> checkpoints) {
     return CompletableFuture.supplyAsync(() -> {
       for (VeniceChangeCoordinate coordinate: checkpoints) {
-        internalSeek(Collections.singleton(coordinate.getPartition()), coordinate.getTopic(), foo -> {
-          // TODO: This is a hack until we refactor out kafkaConsumer, delete this.
+        PubSubTopic topic = pubSubTopicRepository.getTopic(coordinate.getTopic());
+        PubSubTopicPartition pubSubTopicPartition = new PubSubTopicPartitionImpl(topic, coordinate.getPartition());
+        internalSeek(Collections.singleton(coordinate.getPartition()), topic, foo -> {
           Long topicOffset = ((ApacheKafkaOffsetPosition) coordinate.getPosition()).getOffset();
-          kafkaConsumer.seek(new TopicPartition(coordinate.getTopic(), coordinate.getPartition()), topicOffset);
+          pubSubConsumerSeek(pubSubTopicPartition, topicOffset);
         }).join();
       }
       return null;
     });
   }
 
+  private void pubSubConsumerSeek(PubSubTopicPartition topicPartition, Long offset) {
```

Comment:
This looks sketchy to me.  Why not use the pubSubposition in VenicechangeCoordinate?

Reply:
Here the logic is for reusing the -1 logic to offset seek to next in our kafka consumer implementation, it is also called inside  `seekToTail`. In `seekToTail`, the Long offset is from `endOffset` call for the partition. If we convert it to PubSubPosistion and then we still need to covert back to long for subscribe.

Reply:
I wonder if we can do something a bit cleaner here.  My concern is that we're just going to have come back in here when we need to sub in the Xinfra since it won't recognize the long offsets.  Maybe even put it into a set of utility functions that we touch later.

Reply:
I'll leave it to you, if you think this is as surgical as we can by right now I'll defer (it'll be you or @sushantmane who has to come back in here later anyway XD)

Reply:
I would prefer to keep it here, as now the `PubSubConsumerAdapter` is still with Long offset. Even we use pubSubposition in VenicechangeCoordinate, we still need tranlate to Long to interact with  `PubSubConsumerAdapter`.